src/previewSupport/handleConsoleLogs.js

```diff
@@ -0,0 +1,68 @@
+import isString from 'lodash/isString';
+import channel from './channel';
+// eslint-disable-next-line no-eval
+const globalEval = window.eval;
+
+function reducer(accumulator, currentValue) {
+  let evaluatedValue;
+  if (isString(currentValue)) {
+    evaluatedValue = currentValue;
+  } else {
+    evaluatedValue = globalEval(currentValue);
```

Comment:
If I don't do the string check, then eval tries to evaluate whatever string passed to it as code. eval("whatever"); in your console says 'Uncaught ReferenceError: whatever is not defined.'

We need to eval stuff because what if I do console.log(myFunction(6)) or, simpler, console.log(5 * 6).

Reply:
@alexpelan but if you do `console.log(5 * 6)` then our `log` implementation will receive the value `30`, won’t it?

And as you rightly point out, if we were to do `console.log("5 * 6")`, we would want it to print `"5 * 6"`, not `30`.

So I am still not clear on why we want `eval` here at all?

Reply:
hmmmmmmmmm I don't know how I ended up with this impression but I swear at one point our implementation was not receiving 30 and I had to put the eval in to get it to work right. You are correct, we can literally just do a join(" ") on the arguments it appears.